packages/wonder-blocks-icon/components/icon.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import React from "react";
+import {StyleSheet} from "aphrodite";
+import Color from "@khanacademy/wonder-blocks-color";
+
+import addStyle from "../../wonder-blocks-core/util/add-style.js";
+import iconPaths from "../util/icon-paths.js";
+
+type IconName = $Keys<typeof iconPaths>;
+type IconSize = "small" | "medium" | "large" | "xlarge";
+
+type Props = {
+    /**
+     * Passed transparently to the SVG. If not provided, we assume the SVG is
+     * purely decorative and it is invisible to screenreaders.
+     */
+    "aria-label"?: string,
+    /**
+     * The color of the icon. Will default to off-black.
+     */
+    color?: string,
+    /**
+     * One of our named icons from icon-paths.js
+     */
+    name: IconName,
```

Comment:
Relying on a `name` means that we'll have to update `wonder-blocks-icon` to add new icons.  Initiatives should be able to use their own custom icons if need be.  Maybe we could have an `icon` prop or `asset` prop that looks like:
```
{
       small?: string,
       medium?: string,
       large?: string,
       xlarge?: string,
}
```

Reply:
Okay, but if initiatives request initiative specific icons be added to wonder-blocks I think we should say "no".

Reply:
My understanding at the time of submitting this PR is that we didn't actually need to support non-WB icons. That said, I rather like this pattern and I have since refactored the branch to support it. Maybe I should put it up so you can have a look?

Reply:
@idreyn thanks for making this change.  I think the thing we didn't want was having the support the current icon format from webapp, but now we have a new format. 🙌